Plugins/Public/base_plugin/Main.cpp

```diff
@@ -2757,3 +2784,36 @@ EXPORT PLUGIN_INFO* Get_PluginInfo()
 	p_PI->lstHooks.push_back(PLUGIN_HOOKINFO((FARPROC*)&Plugin_Communication_CallBack, PLUGIN_Plugin_Communication, 11));
 	return p_PI;
 }
+
+void ResetShieldStrength() {
+	for (map<uint, PlayerBase*>::iterator i = player_bases.begin(); i != player_bases.end(); ++i) {
+		i->second->ResetShieldStrength();
+	}
+}
+
+bool checkBaseVulnerabilityStatus() {
+
+	if (baseVulnerabilityWindows.empty()) {
+		return false;
+	}
+
+	time_t tNow = time(0);
+	struct tm *t = localtime(&tNow);
+	uint currHour = t->tm_hour;
+	// iterate over configured vulnerability periods to check if we're in one.
+	// - in case of timeStart < timeEnd, eg. 5-10, the base will be vulnerable between 5AM and 10AM.
+	// - in case of timeStart > timeEnd, eg. 23-2, the base will be vulnerable after 11PM and before 2AM, so a period of 3 hours.
+	for (list<BASE_VULNERABILITY_WINDOW>::iterator i = baseVulnerabilityWindows.begin(); i != baseVulnerabilityWindows.end(); ++i){
+		if((i->start < i->end 
+			&& i->start <= currHour && i->end > currHour)
+		|| (i->start > i->end
+			&& (i->start <= currHour || i->end > currHour))) {
```

Comment:
should we add a condition here for `i->start == i->end` ? otherwise I think a bad config could make future base vulnerability windows not count - they would be vulnerable all the time as this code would fail and it'd fall through to the default return

Reply:
If config would contain a `i->start == i->end` entry, it'd skip the `baseVulnerabilityWindows.empty()` check, and fail the highlighted check, resulting in all bases being permanently invulnerable.
The only way to make bases permanently vulnerable would be to either configure no vulnerability entries, or have entry/entries spanning the entire 24 hour range.

Reply:
good point, not sure what I was thinking when I wrote this